### PR TITLE
Docs/1013 select option api

### LINKED ISSUE #1013: [Docs?]: Select Option API Documentation
### Latest version

- [X] I have checked the latest version

### Summary 💡

Add documentation for new `getSelectOptionProps` and `getJSSelectOptionProps` APIs

### Motivation 🔦

_No response_

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

closes #1013 

## What is the new behavior?
Updates `Select` documentation to include `SelectOption`

## Other information
Noticed inconsistency in JS API output.  Opened #1026.  Documentation is for the API as it will be after that is fixed.

### CODE REVIEW

File: website/docs/development/headless-styles/Select.mdx [hunk 1]
```diff
@@ -81,8 +84,8 @@ A Select should only ever be `invalid` when it is required and an empty "placeho
 
 ```javascript
```
Comment (caseybaggz) [inline website/docs/development/headless-styles/Select.mdx:85]:
```suggestion
```javascript showLineNumbers
```

File: website/src/components/Select/Select.js [hunk 2]
```diff
@@ -33,14 +45,10 @@ function Select(props) {
       <div {...selectProps.selectWrapper}>
         <select {...selectProps.select} onChange={props.onChange}>
           {props.placeholder && (
-            <option hidden disabled selected value="">
-              {props.placeholder}
-            </option>
+            <Option hidden disabled value="" label={props.placeholder} />
```
Comment (caseybaggz) [inline website/src/components/Select/Select.js:48]: I think there was a big opportunity missed in this API to not support the initial Option item. 🤔

**Review by caseybaggz — DISMISSED** [review]

File: website/src/components/Select/Select.js (hunk 2 above)
Comment (Rykus0) [reply website/src/components/Select/Select.js:48]: ah, I hadn't thought of that.  I know there's some debate about whether or not they should be used, but they're pretty commonplace.  I can open an issue to add it as a 'kind' option.

Comment (Rykus0) [reply website/src/components/Select/Select.js:48]: Opened #1030 for this

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**